### PR TITLE
LLM: Fix discards in `optimize_model` with non-hf models  and add openai whisper example

### PR DESCRIPTION
**Pull request by leonardozcm**
## Description

Fix the requirement for the source of the model in `optimize()`, for it strongly ask for model's `.config` attribute which only exits in huggingface transformer models.
And add new an exmaple with openai-tiny.

### CODE REVIEW

File: python/llm/example/pytorch-model/openai-whisper/recognize.py [hunk 1]
```diff
@@ -0,0 +1,58 @@
+import whisper
```
Comment (plusbang) [inline python/llm/example/pytorch-model/openai-whisper/recognize.py:1]: This line maybe removed to L18?

**Review by plusbang — APPROVED** [review]
Other LGTM

File: python/llm/example/pytorch-model/openai-whisper/recognize.py [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+    target_sr = 16000
+    audio = librosa.resample(y,
+                            orig_sr=sr,
+                            target_sr=target_sr)
+
+    # Load whisper model under pytorch framework
+    model = whisper.load_model(args.model_name)
+
+    # With only one line to enable bigdl optimize on a pytorch model
+    model = optimize_model(model, low_bit="sym_int4", optimize_llm=True)
```
Comment (jason-dai) [inline python/llm/example/pytorch-model/openai-whisper/recognize.py:50]: optimize_llm should default to true

File: python/llm/src/bigdl/llm/transformers/convert.py [hunk 3]
```diff
@@ -114,7 +115,11 @@ def ggml_convert_quant(model, qtype, optimize_model=True, device="cpu"):
         pass
 
     if optimize_model:
-        model = optimize(model)
+        try:
+            model = optimize(model)
+        except:
+            logger.info("It is not supported that optimizing a model isn't belong to"
```
Comment (jason-dai) [inline python/llm/src/bigdl/llm/transformers/convert.py:121]: what do you mean?

Comment (leonardozcm) [reply python/llm/src/bigdl/llm/transformers/convert.py:121]: `optimize()`  is only compatible with the Hugging Face API, and take effects on `chatglm` and `llama` family more specifically.
We apply `optimize()` on other pytorch model is likely to fail, `optimize()` strongly assume it is [a hf format model](https://github.com/intel-analytics/BigDL/blob/main/python/llm/src/bigdl/llm/transformers/convert.py#L142):
```python
if "chatglm2" in model.config._name_or_path:
```

Do we need to explore a way more general to extend its working range?

Comment (leonardozcm) [reply python/llm/src/bigdl/llm/transformers/convert.py:121]: I see

Comment (jason-dai) [reply python/llm/src/bigdl/llm/transformers/convert.py:121]: If there is any issue, you should fix `optimize` instead

File: python/llm/src/bigdl/llm/transformers/convert.py [hunk 4]
```diff
@@ -129,6 +130,14 @@ def convert_forward(m, target_m, new_forward):
 def optimize(model):
     from packaging import version
     from bigdl.llm.transformers.models.llama import llama_attention_forward_4_31
+    from transformers.modeling_utils import PreTrainedModel
+
+    # All huggingface format models are inherited from `PreTrainedModel`
+    if not isinstance(model, PreTrainedModel):
+        logger.info("It is not supported that optimizing a model isn't belong to"
+                    "huggingface transformers with `optimize_model=True` for now.")
```
Comment (jason-dai) [inline python/llm/src/bigdl/llm/transformers/convert.py:138]: Only HuggingFace Transformers models are currently supported for further optimizations

**Pull request merged**